src/candid_rpc.rs

```diff
@@ -27,34 +30,43 @@ impl RpcTransport for CanisterTransport {
         METADATA.with(|m| m.borrow().get().nodes_in_subnet)
     }
 
-    fn resolve_api(provider: RpcNodeProvider) -> Result<RpcApi, ProviderError> {
-        // TODO: https://github.com/internet-computer-protocol/ic-eth-rpc/issues/73
-        Ok(provider.api())
+    fn resolve_api(provider: &RpcNodeProvider) -> Result<RpcApi, ProviderError> {
+        use RpcNodeProvider::*;
+        let (chain_id, hostname) = match provider {
+            Ethereum(provider) => (
+                ETH_MAINNET_CHAIN_ID,
+                match provider {
+                    EthereumProvider::Ankr => "rpc.ankr.com",
+                    EthereumProvider::BlockPi => "ethereum.blockpi.network",
+                    EthereumProvider::Cloudflare => "cloudflare-eth.com",
+                },
+            ),
+            Sepolia(provider) => (
+                ETH_SEPOLIA_CHAIN_ID,
+                match provider {
+                    SepoliaProvider::Ankr => "rpc.ankr.com",
+                    SepoliaProvider::BlockPi => "ethereum-sepolia.blockpi.network",
+                    SepoliaProvider::PublicNode => "ethereum-sepolia.publicnode.com",
+                },
+            ),
+        };
+        Ok(
+            find_provider(|p| p.chain_id == chain_id && p.hostname == hostname)
+                .ok_or(ProviderError::ProviderNotFound)?
+                .api(),
+        )
     }
 
-    async fn call_json_rpc<T: DeserializeOwned>(
-        provider: RpcNodeProvider,
-        json: &str,
-        max_response_bytes: u64,
-    ) -> Result<T, RpcError> {
-        let response = do_http_request(
-            ic_cdk::caller(),
-            ResolvedSource::Api(Self::resolve_api(provider)?),
-            json,
-            max_response_bytes,
+    async fn http_request(
+        _provider: &RpcNodeProvider,
```

Comment:
Why is the provider ignored?

Reply:
The `provider` param will eventually be used for logging. For context, the `http_request()` method was originally added to enable mock HTTP outcalls in unit tests, but I removed this in favor of using state machine tests.